File: accidentes/AMarotoChivite-Accidentes/AMarotoChivite-Accidentes-Java/src/main/java/Ficheros/Accidentes/models/dto/AccidenteDto.java
package Ficheros.Accidentes.models.dto;

import com.squareup.moshi.Json;
import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;

import java.util.Collection;

public class AccidenteDto {
    @Attribute(name = "num_expediente") // Para XML
    private final String numExpediente;
    @Attribute(name = "fecha")
    private final String fecha;
    @Element(name = "hora")
    private final String hora;
    @Attribute(name = "localizacion")
    private final String localizacion;
    @Attribute(name = "numero")
    private final String numero;
    @Element(name = "cod_distrito")
    private final String codDistrito;
    @Element(name = "distrito")
    private final String distrito;
    @Element(name = "tipo_accidente")
    private final String tipoAccidente;
    @Element(name = "estado_meteorologico")
    private final String estadoMeteorologico;
    @Element(name = "tipo_vehiculo")
    private final String tipoVehiculo;
    @Element(name = "tipo_persona")
    private final String tipoPersona;
    @Element(name = "rango_edad")
    private final String rangoEdad;
    @Element(name = "sexo")
    private final String sexo;
    @Element(name = "cod_lesividad")
    private final String codLesividad;

    @Element(name = "lesividad")
    private final String lesividad;

    @Element(name = "coordenada_x")
    private final String coordenadaX;

    @Element(name = "coordenada_y")
    private final String coordenadaY;

    @Element(name = "positiva_alcohol")
    private final String positivaAlcohol;

    @Element(name = "positiva_droga")
    private final String positivaDroga;

    public AccidenteDto(
            @Attribute(name = "num_expediente") String numExpediente,
            @Attribute(name = "fecha") String fecha,
            @Element(name = "hora") String hora,
            @Attribute(name = "localizacion") String localizacion,
            @Attribute(name = "numero") String numero,
            @Element(name = "cod_distrito") String codDistrito,
            @Element(name = "distrito") String distrito,
            @Element(name = "tipo_accidente") String tipoAccidente,
            @Element(name = "estado_meteorologico") String estadoMeteorologico,
            @Element(name = "tipo_vehiculo") String tipoVehiculo,
            @Element(name = "tipo_persona") String tipoPersona,
            @Element(name = "rango_edad") String rangoEdad,
            @Element(name = "sexo") String sexo,
            @Element(name = "cod_lesividad") String codLesividad,
            @Element(name = "lesividad") String lesividad,
            @Element(name = "coordenada_x") String coordenadaX,
            @Element(name = "coordenada_y") String coordenadaY,
            @Element(name = "positiva_alcohol") String positivaAlcohol,
            @Element(name = "positiva_droga") String positivaDroga
    ) {
        this.numExpediente = numExpediente;
        this.fecha = fecha;
        this.hora = hora;
        this.localizacion = localizacion;
        this.numero = numero;
        this.codDistrito = codDistrito;
        this.distrito = distrito;
        this.tipoAccidente = tipoAccidente;
        this.estadoMeteorologico = estadoMeteorologico;
        this.tipoVehiculo = tipoVehiculo;
        this.tipoPersona = tipoPersona;
        this.rangoEdad = rangoEdad;
        this.sexo = sexo;
        this.codLesividad = codLesividad;
        this.lesividad = lesividad;
        this.coordenadaX = coordenadaX;
        this.coordenadaY = coordenadaY;
        this.positivaAlcohol = positivaAlcohol;
        this.positivaDroga = positivaDroga;
    }

    public String getPositivaDroga() {
        return positivaDroga;
    }

    public String getPositivaAlcohol() {
        return positivaAlcohol;
    }

    public String getDistrito() {
        return distrito;
    }

    public String getSexo() {
        return sexo;
    }

    public String getFecha() {
        return fecha;
    }

    public String getTipoVehiculo() {
        return tipoVehiculo;
    }

    public String getLocalizacion() {
        return localizacion;
    }

    public String getHora() {
        return hora;
    }
    public String getLesividad() {
        return lesividad;
    }

    public String getEstadoMeteorologico() {
        return estadoMeteorologico;
    }

    public String getTipoAccidente() {
        return tipoAccidente;
    }

    @Override
    public String toString() {
        return "AccidenteDto{" +
                "numExpediente='" + numExpediente + '\'' +
                ", fecha='" + fecha + '\'' +
                ", hora='" + hora + '\'' +
                ", localizacion='" + localizacion + '\'' +
                ", numero='" + numero + '\'' +
                ", codDistrito='" + codDistrito + '\'' +
                ", distrito='" + distrito + '\'' +
                ", tipoAccidente='" + tipoAccidente + '\'' +
                ", estadoMeteorologico='" + estadoMeteorologico + '\'' +
                ", tipoVehiculo='" + tipoVehiculo + '\'' +
                ", tipoPersona='" + tipoPersona + '\'' +
                ", rangoEdad='" + rangoEdad + '\'' +
                ", sexo='" + sexo + '\'' +
                ", codLesividad='" + codLesividad + '\'' +
                ", lesividad='" + lesividad + '\'' +
                ", coordenadaX='" + coordenadaX + '\'' +
                ", coordenadaY='" + coordenadaY + '\'' +
                ", positivaAlcohol='" + positivaAlcohol + '\'' +
                ", positivaDroga='" + positivaDroga + '\'' +
                '}';
    }

}

File: accidentes/AMarotoChivite-Accidentes/AMarotoChivite-Accidentes-Kotlin/src/main/kotlin/Ficheros/Accidentes/models/dto/AccidenteDTO.kt
package Ficheros.Accidentes.models.dto

import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element

class AccidenteDTO(
    @field:Attribute(name = "num_expediente")
    @param:Attribute(name = "num_expediente")
    val numExpediente: String,
    @field:Attribute(name = "fecha")
    @param:Attribute(name = "fecha")
    val fecha: String,
    @field:Element(name = "hora")
    @param:Element(name = "hora")
    val hora: String,
    @field:Attribute(name = "localizacion")
    @param:Attribute(name = "localizacion")
    val localizacion: String,
    @field:Attribute(name = "numero")
    @param:Attribute(name = "numero")
    val numero: String,
    @field:Element(name = "cod_distrito")
    @param:Element(name = "cod_distrito")
    val codDistrito: String,
    @field:Element(name = "distrito")
    @param:Element(name = "distrito")
    val distrito: String,
    @field:Element(name = "tipo_accidente")
    @param:Element(name = "tipo_accidente")
    val tipoAccidente: String,
    @field:Element(name = "estado_meteorologico")
    @param:Element(name = "estado_meteorologico")
    val estadoMeteorologico: String,
    @field:Element(name = "tipo_vehiculo")
    @param:Element(name = "tipo_vehiculo")
    val tipoVehiculo: String,
    @field:Element(name = "tipo_persona")
    @param:Element(name = "tipo_persona")
    val tipoPersona: String,
    @field:Element(name = "rango_edad")
    @param:Element(name = "rango_edad")
    val rangoEdad: String,
    @field:Element(name = "sexo")
    @param:Element(name = "sexo")
    val sexo: String,
    @field:Element(name = "cod_lesividad")
    @param:Element(name = "cod_lesividad")
    val codLesividad: String,
    @field:Element(name = "lesividad")
    @param:Element(name = "lesividad")
    val lesividad: String,
    @field:Element(name = "coordenada_x")
    @param:Element(name = "coordenada_x")
    val coordenadaX: String,
    @field:Element(name = "coordenada_y")
    @param:Element(name = "coordenada_y")
    val coordenadaY: String,
    @field:Element(name = "positiva_alcohol")
    @param:Element(name = "positiva_alcohol")
    val positivaAlcohol: String,
    @field:Element(name = "positiva_droga")
    @param:Element(name = "positiva_droga")
    val positivaDroga: String
) {
    override fun toString(): String {
        return "AccidenteDTO(numExpediente='$numExpediente', fecha='$fecha', hora='$hora', localizacion='$localizacion', numero='$numero', codDistrito='$codDistrito', distrito='$distrito', tipoAccidente='$tipoAccidente', estadoMeteorologico='$estadoMeteorologico', tipoVehiculo='$tipoVehiculo', tipoPersona='$tipoPersona', rangoEdad='$rangoEdad', sexo='$sexo', codLesividad='$codLesividad', lesividad='$lesividad', coordenadaX='$coordenadaX', coordenadaY='$coordenadaY', positivaAlcohol='$positivaAlcohol', positivaDroga='$positivaDroga')"
    }
}
